spectrocloud/resource_cluster_vsphere.go

```diff
@@ -452,6 +455,19 @@ func resourceClusterVsphereUpdate(ctx context.Context, d *schema.ResourceData, m
 
 	if d.HasChange("machine_pool") {
 		oraw, nraw := d.GetChange("machine_pool")
+		if oraw != nil && nraw != nil {
```

Comment:
Let's have a unit test to cover this update function?

Reply:
As discussed earlier, we currently don't have support to simulate state change (has change for particular attribute ) in terraform SDK through the unit test. we can't cover it in the unit test for now